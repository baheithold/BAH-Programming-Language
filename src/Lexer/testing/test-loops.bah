# This file tests valid loops


for (var i = 0; i < length; i++) {
    #do something
}

while (i < length) {
    # do something
}
