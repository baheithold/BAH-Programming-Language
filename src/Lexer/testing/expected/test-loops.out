FOR_KEYWORD
OPAREN
VAR_KEYWORD
ID_TYPE i
ASSIGN_BINARY
INTEGER_TYPE 0
SEMICOLON
ID_TYPE i
LESSER_THAN_COMPARATOR
ID_TYPE length
SEMICOLON
ID_TYPE i
INCREMENT_UNARY
CPAREN
OBRACE
CBRACE
WHILE_KEYWORD
OPAREN
ID_TYPE i
LESSER_THAN_COMPARATOR
ID_TYPE length
CPAREN
OBRACE
CBRACE
END_OF_INPUT
